pkg/config/config.go

```diff
@@ -94,6 +98,28 @@ type GroupVersionKind struct {
 	Kind    string `toml:"kind,omitempty"`
 }
 
+// PromptConfig represents an inline prompt definition in TOML
+type PromptConfig struct {
+	Name        string                 `toml:"name"`
+	Title       string                 `toml:"title,omitempty"`
+	Description string                 `toml:"description"`
+	Arguments   []PromptArgumentConfig `toml:"arguments,omitempty"`
+	Messages    []PromptMessageConfig  `toml:"messages"`
+}
+
+// PromptArgumentConfig represents a prompt argument in TOML
+type PromptArgumentConfig struct {
+	Name        string `toml:"name"`
+	Description string `toml:"description"`
+	Required    bool   `toml:"required"`
+}
+
+// PromptMessageConfig represents a prompt message in TOML
+type PromptMessageConfig struct {
+	Role    string `toml:"role"`
+	Content string `toml:"content"`
+}
+
```

Comment:
These structs are exactly the same as those defined in the api package, is there any reasoning to duplicate them?

I see we've created multiple versions of the same data stuctures that could easily be consolidated while preserving their single responsibility.

I believe the pkg/api/prompts.go could contain could define the single struct with all the applicable tags (`json`, `yaml`, `toml`, etc) and be reused everywhere else.

> ### Scope
> - Define prompt data structures in pkg/api/ (Prompt, PromptArgument, PromptMessage, PromptContent)

_[[CONFIG] Configurable prompt templates #226](https://github.com/containers/kubernetes-mcp-server/issues/226)_

Reply:
reworked the types to match the requirements and simplify